components/eamxx/tests/postrun/check_hashes_ers.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
```

Comment:
Would be nice to check this script in our nightlies. I'm not really sure how to do it though. I guess we'd have to introduce some bug in some atm proc, to trigger a diff in the hash. Perhaps not too important, unless @jgfouca has some clear idea of how to do it (and assuming it's not too time consuming, and doesn't require tons of additions in the code).

Reply:
I may not follow, but to be clear, this script will run at the end of the RUN phase for ERS_Ln22.ne30_ne30.F2010-SCREAMv1.scream-internal_diagnostics_level in our nightlies as a result of the POSTRUN_SCRIPT setting in the internal_diagnostics testmod. If the script detects no hash diffs, it will return 0, permitting the RUN phase to pass. If it detects diffs, it will return non-0, forcing the RUN phase to fail.

Reply:
What I meant is to have a way to ensure that this script works. I was thinking of having an ERS test where the hash _does_ change, and verify that the script indeed triggers a fail. Along the line of what we already do in our scripts-tests for standalone tests.

But as I said, this is not too important. I'm assuming you already checked the script does fail when it's supposed to anyways.

Reply:
Yes. One easy way to make it fail is to exclude `atmosphere_processes::internal_diagnostics_level=0` from the `atmchange` command. Something in the "EAMxx" process is including buffers that aren't actually filled, leading to run-to-run diffs at every time step. I found and fixed a similar problem in the Homme AD interface but haven't looked at the EAMxx process.

Reply:
This isn't what you were asking for, but I was reminded that I wanted to separate out and unit test the core hashing routines, which I've now done.